\documentclass[11pt, a4paper,oneside]{report}
\usepackage[left=3cm,right=3cm,top=3cm,bottom=3cm]{geometry}
\usepackage{setspace}
\onehalfspacing

\newcommand{\mychapter}[2]{
    \setcounter{chapter}{#1}
    \setcounter{section}{0}
    \chapter*{#2}
    \addcontentsline{toc}{chapter}{#2}
}
\usepackage{verbatim}
\usepackage{graphicx}
\usepackage{chngcntr}
\counterwithout{figure}{chapter}
\usepackage{multirow}
\usepackage{subfigure}
\usepackage{hyperref}

\usepackage{float}

\begin{document}

%\title{Research plan}
%\subtitle{Evolutionary innovation in eukaryotic proteins}
%\author{Oxana Sachenkova}
%\date{2013}
%\emph{Supervisor:} Arne Elofsson
\begin{titlepage}

\newcommand{\HRule}{\rule{\linewidth}{0.5mm}} % Defines a new command for the horizontal lines, change thickness here

\center % Center everything on the page
 
%----------------------------------------------------------------------------------------
%	HEADING SECTIONS
%----------------------------------------------------------------------------------------
 
\textsc{\Large Research plan}\\[0.5cm]  

%----------------------------------------------------------------------------------------
%	TITLE SECTION
%----------------------------------------------------------------------------------------

\HRule \\[0.4cm]
{ \huge \bfseries Evolutionary innovation in eukaryotic proteins}\\[0.4cm] % Title of your document
\HRule \\[1.5cm]
 
%----------------------------------------------------------------------------------------
%	AUTHOR SECTION
%----------------------------------------------------------------------------------------

\begin{minipage}{0.4\textwidth}
\begin{flushleft} \large
\emph{Author:}\\
Oxana \textsc{Sachenkova}  
\end{flushleft}
\end{minipage}
~
\begin{minipage}{0.4\textwidth}
\begin{flushright} \large
\emph{Supervisor:} \\
Prof. Arne \textsc{Elofsson} % Supervisor's Name
\end{flushright}
\end{minipage}\\[4cm]

  
%----------------------------------------------------------------------------------------
%	DATE SECTION
%----------------------------------------------------------------------------------------

{\large \today}\\[3cm] % Date, change the \today to a set date if you want to be precise

 
\vfill % Fill the rest of the page with whitespace

\end{titlepage}

\mychapter{0}{Introduction}
More than 150 years after the publication of Darwin's {\itshape Origin of Species}, we still struggle to understand all details of the evolutionary processes and how they shape the diversity of life on Earth. The application of the idea of "descent with modification" to the genome continues to produce fundamental insights into how biological systems evolve. Changes at the DNA level that might occur due to recombination events, errors during meiosis and random mutations can alter the proteins synthesized in a cell. Those changes can affect gene expression levels, protein structure and function, the way proteins interact. Analysis of protein evolution is essential for investigating important biological questions such as identification of functionally important sites, finding potential drug targets and understanding how proteins function.
  
Advances in experimental methods such as whole-genome sequencing, RNA sequencing and mass spectrometry generate vast amounts of quality DNA and protein sequence data.  With the costs of such experiments going down, data for different types of cells, tissues and organisms becomes available. This large increase of genome-wide information provides new opportunities to study molecular evolution using computational and statistical approaches.

In my research I'm focusing on studying the mechanisms of generating novel protein structures and functions. The interplay between gene duplication, insertion and deletion events, mutations and evolutionary constraints can lead to the appearance of new structural elements,  
protein function divergence and protein length variation. By analyzing protein and DNA sequences we can infer evolutionary histories and study the impact that changes in the sequence have on protein function and structure.  More detailed research questions on which I'm concentrating my efforts include: How do long indels and duplications influence protein function? How do these processes differ in case of naturally unstructured proteins? 

\mychapter{1}{Background}

This chapter introduces the key concepts that are important for understanding the proposed research topic. 
\section{Basic terminology}
\subsection{Building blocks of life}
%definition of a cell and dna
%TODO: perhaps write more details about what RNA is
A cell is the smallest unit of living matter, it requires energy to maintain its organization and is capable of reproducing itself. These properties are central to the definition of life. All known living organisms consist of one or multiple cells, and in all cases, the whole organism is generated by cell divisions from a single cell. The single cell, therefore, contains all the necessary information and machinery that defines the organism.  Since the discovery by Watson and Crick in 1953 \cite{Watson1974} we know that all the living cells store this information in the form of double-stranded DNAs -  long molecular chains, formed always of the same subunits, nucleic acids (nucleotides). Nucleotides include four different types of organic molecules: adenine, guanine, cytosine and thymine (also referred to as A, G, C and T). These nucleic base pairs are attached  in a linear sequence that encodes the hereditary information of the cell. Information stored in DNA is read out and put to use through a two step process: first, in transcription, segments of DNA sequence are used to guide the synthesis of many RNA molecules, then, in translation some of the RNA molecules are used to synthesize protein molecules while some are helping in this process.

% translation rules, codons, definition of a gene and genome
 A certain set of rules for translating DNA to protein is shared between all living cells. Different combinations of three base pairs in the DNA, named codons, code for a certain amino acid or serve as a signal of a start or end of a polypeptide. This genetic code rule set is redundant, i.e, some codons code for the same amino acid \cite{Turanov2009}. Each segment of DNA that encodes for a functional molecule, either RNA or protein is called a gene\cite{Gerstein2007}. Regulatory elements, - sequences that influence the gene transcription processes, sometimes are also considered as an integral part of a gene \cite{lodish2001}. The complete set of DNA molecules in an organism, including all of its genes is called a genome.  
 
%definition of a protein
Proteins are long unbranched polymer chains, formed by different monomeric building units, amino acids. Each protein molecule, or polypeptide, is created by joining amino acids in a particular sequence, which then folds into a three-dimensional form. Proteins constitute more than 50\%  of the cell's dry mass \cite{Alberts2007} and perform nearly all functions in a cell: catalyzing metabolic reactions, replicating DNA, responding to stimuli, and transporting molecules from one location to another.
\begin{figure}[t]
\begin{center}
\label{img:dnatoprotein}
\includegraphics[width=0.8\textwidth]{figures/dna_to_protein.jpg}
\end{center}
\caption{An overview of the DNA to protein translation process. 1. RNA polymerase transcribes DNA to make messenger RNA (mRNA). 2. The mRNA sequence (dark red strand) is complementary to the DNA sequence (blue strand). 3. On ribosomes, transfer RNAs (tRNA) help convert mRNA into protein. 4. Amino acids link up to make a protein.} 
\end{figure}

\subsection{Three domains of life}
%3 domains of life
Considering their evolutionary descent and biochemical properties living organisms are classified into three main branches: Archaea, Eukarya and Bacteria. In Archaea and Bacteria DNA is not separated from the cytoplasm. Genetic material in Eukarya, in contrast, is packed in a unique organelle called the nucleus. Archae is evolutionary equally distant from Bacteria and Eukarya and possess unique lipid compositions and translational mechanisms. 

%further subdivision of eukarya
Most known organisms of Bacteria and Archaea are single-celled, while eukaryotes can be unicellular or consist of many cells of different type and function. Eukarya is further subdivided into five kingdoms: Protozoa (unicellular eukaryotic organisms), Chromista (algae, simple organisms with chloroplasts), Plantae (plants), Fungi (yeast, molds and mushrooms) and Animalia (animals).    

%genome size and molecular evolution differences
The three domains of life demonstrate fundamental differences in genome organization and mechanisms of evolution. The genome of Bacteria is limited to 180kb - 13 mb and Archaea to 500kb - 5 mb \cite{Koonin2008}, while that of eukaryotes can be characterized by a significant expansion in size and complexity \cite{Gregory2007, Parfrey2008}. Evolution of bacterial and archaea genomes involves two major processes: acquisition of exogenous DNA through horizontal gene transfer and genome decay through deletion \cite{Pal2005}. The genomes of Archaea are similar to those of Bacteria in size and gene density, but some translational mechanisms are the same as those in eukaryotes.

% what's special about Eukarya?
 Eukaryotes demonstrate a significant increase in genetic information compared to the other two domains of life. An abundance of non-coding DNA, gene duplications and alternative splicing create more possibilities for complicated evolutionary pathways in eukaryotic genomes.
 
  In my research I'm focusing on studying the evolution of eukaryotic proteins. In the following chapters the mechanisms of evolution in eukaryotic genomes are described in detail.
  
\subsection{Modern Theory of Evolution}
% short about history
Evolutionary theory famously dates back to 1858 when maybe the most influential papers in biology were published. 
Darwin and Wallace independently presented the hypothesis of “descent with modification” that accounts for the diversity 
and appearance of life on Earth \cite{Wallace1912}. Advances in the fields of population genetics, paleontology, developmental and 
molecular biology formed the integrated view of evolution, largely developed in 1930-50s by Fisher, Mayr and others \cite{MAYR1963,Grant1980,Kutschera2004}. 

%what is evolution? what are species? 
According to the modern views, evolution is the change in the inherited characteristics in interbreeding populations of 
living organisms, species. Genetic and phenotypic variety in these populations is introduced through mutations, changes in the DNA sequence of an organism. In sexually reproducible organisms another source of variation comes into play - recombination, an exchange of genetical information between two organisms. The amount of variation that is passed from one generation to the next is quite small; almost all variations will be lost or be neutral (not affect the fitness of the organism). However, some of the changes in hereditary material can lead to changes in the phenotype, physical characteristics of an organism, that in turn affect the survival of the organism and its ability to produce offspring. 

%forces of evolution: natural selection, selective pressures
Natural selection is the process that determines what characteristics will persist in a population, based on the reproductive fitness of the individuals in this population. In a competition over the limited resources only organisms that can adapt to the changes in the environment and produce successful offspring will survive. Potential selective pressures that affect survival include the availability of prey, competition with the other organisms (or other species) over food and reproduction partners, climate and other habitat conditions changes. An example of an adaptive process shaping the evolution of the population is represented in the Figure 2.

%population size influence
Population size is another factor that affects the selection process.  In small, reproductively isolated populations, for example, special circumstances exist that can produce rapid random changes in genetic material. The smaller the population, the more susceptible it is to random changes in genetic variants frequencies.  This phenomenon is known as genetic drift. Immigration of individuals from other populations can also influence changes in the phenotype. This process is called gene flow \cite{Grant1980}.

\begin{figure}[t]
\begin{center}
\label{img:populations}
\includegraphics[width=0.7\textwidth]{figures/squirrel.jpg}
\end{center}
\caption{ Illustration of the evolutionary process in the Eastern Grey Squirrel population. There are two phenotypes, grey and black fur, represented with green and orange circles respectively. The grey type, which is dominant in Generation N, has been gradually replaced by the black type by Generation N + 2 due to a selective advantage: black is an efficient camouflage against their main predators.} 
\end{figure}

In the early 1960s the first sequencing data became available from different groups of organisms. It was only a few small proteins, such as insulins \cite{Sanger1945}, hemoglobins \cite{INGRAM1956} and ribonuclease \cite{Kartha1967}, but already then it became apparent that the evolution on the molecular level cannot only be explained by the adaptive processes and the mechanisms of introducing variety in the genome are not limited to mutation and recombination. A new field, molecular evolution emerged and new theories to explain the underlying processes of phenotypic diversity and genome size evolution were proposed.  In the next chapter I will describe the mechanisms of evolution on the molecular level with focus on protein coding genes. 


\section{Evolution of protein coding genes}
%adaptive processes are not enough
 Approaching evolutionary questions from the genotype perspective lead to the emergence of new theories of evolutionary mechanisms. Adaptive evolution driven by natural selection could not explain the overwhelming body of evidence for changes in the aminoacid sequences that don't necessarily change the protein function, mutation rates on the sequence level and increase in genome size and complexity in high order organisms. 
 
 %what is neutral theory?
 Kimura (1968) and King and Jukes (1969) proposed that most mutant substitutions at the molecular level must be selectively neutral or nearly neutral, which created the alternative theory of neutral evolution \cite{Kimura1968,King1969}. It has been controversial ever since, however, as more data becomes available from whole-genome sequencing of various species, it becomes clear that the general pattern of molecular evolution agrees with the neutral theory, even though some studies still suggest there are exceptions to this
rule \cite{Haygood2007,Nielsen2007,Akey2009}  
%need more detail on neutral evolution here

% we don't go into regulatory elements
Our knowledge about gene expression regulation in the genome has expanded significantly in the past 20 years. Since the landmark discovery of RNA interference, a process of small strands of RNA influencing gene expression \cite{Fire1998}, many other regulatory mechanisms have been discovered.  The ENCODE consortium aiming to create a comprehensive encyclopedia of gene regulatory elements reports that even by the most conservative estimates at least 8.5\% of the bases in human genome are involved in direct gene 
regulation \cite{Bernstein2012}. It is important, therefore, to study protein evolution in consideration of the regulatory elements in non-coding regions, non-coding RNAs and other influences on gene expression. However, because the function of most non-coding regions is still poorly understood and the experimental data is sparse, in this research project we will focus mainly on the evolution of protein-coding genes. 

Next I introduce some of the main patterns of protein evolution and the mechanisms of producing variety in the eukaryotic genome. 

\subsection{Protein sequence variations}
\subsubsection{Mutations}
Mutations are sequence changes in DNA and one of the main origins for genetic diversity. For mutations to affect  organism's offspring, they must occur in cells that produce the next generation (as opposed to somatic mutations) and affect the hereditary material. In the smallest type of mutation event a single base pair in a DNA sequence is changed into another base pair. This type of mutations can lead to changes in the protein sequence if this segment or be synonymous, i.e., code for the same amino acid. Synonymous mutations exist because many amino acids are encoded by multiple codons, as described earlier.

% Mutation rate in proteins and population size????
The frequency with which a certain type of mutation occurs is generally expressed as the number of mutations per biological unit (i.e., per cell division, per gamete, or per round of replication).

Based on the effect of the mutation on fitness of the organism it can be beneficial, harmful or neutral. Beneficial or harmful mutations are what allow us to see natural selection in action. Mutations that increase the fitness of the organism and become heritable tend to increase in frequency in the population; harmful, or deleterious mutations, on the other hand, become extinct or stay at lower prevalence in the population. Neutral mutations don't have any effect on fitness and can become widely spread in a population and, according to the neutral theory, the majority of mutations that persist in the population (become fixed) are neutral. 

Mutations may also take the form of insertions or deletions, which together are known as indels. The number of nucleotide bases that is inserted or deleted from the DNA sequence can vary a lot. Indels of one or two bases in the coding sequences are the most common and can have a substantial impact on protein translation by causing a frame shift because of the triplet nature of codons.  

 Short indels that don't cause a frame shift in translation and become fixed in the population often occur in exposed loop regions \cite{Kim2010}. Longer indel events might involve an insertion or deletion of an entire protein domain. The selective pressure acting on these events is less well understood, but fixed long indels are often associated with functional change \cite{Pascual-Garcia2010}.

\subsubsection{Gene duplication}
Gene duplication is agreeably the most important process for the generation of new genes during molecular evolution. During protein evolution analysis it is important to distinguish between the proteins that are derived from a single ancestral gene in the last common ancestor of the given two species, i.e., as a result of a speciation event (orthologs), and proteins that evolved through duplication within the same (perhaps ancestral) genome, which are referred to as paralogs\cite{Jensen2001}.

There are at least two different ways in which it can occur: 
\begin{itemize} 
\item By duplication of a single gene or a group of genes 
\item By whole genome duplication 
\end{itemize}

The first of these mechanisms is the most common: multigene families are widely spread in all known species. By comparing the sequences of the members of these families it is possible to trace individual duplications that occurred in a common ancestral genome. Several mechanisms of these gene duplications are known in eukaryotes: unequal crossing-over, unequal sister chromatid exchange and replication slippage. The initial result of such an event is two identical genes. Selective constraints will ensure that one of these copies is still capable of maintaining the original function; the daughter copy in the meantime is subject to relaxed evolutionary pressure.

The majority of these excess gene copies acquire deleterious mutations that inactivate them so they become pseudogenes \cite{Zhang2001}. Occasionally, though, the mutations that accumulate in this gene copy lead to a new function or a slightly altered sub-function of a parent gene.  Genes coding for different types of globin proteins that bind oxygen, evolved from a common ancestral gene which, after successive duplications and speciation events, led to the genes that encode the widespread globin superfamily \cite{Hardison1998}

Whole Genome Duplication (WGD) is the most rapid mechanism of increasing gene copy number and can occur due to an error during meiosis. The evidence from both experimental and computational studies suggests that many gene innovations in yeast and animals are the results of the WGDs during their evolutionary history \cite{Dehal2005}. Increased complexity in the vertebrate developmental regulatory network has been reported to be the result of the two rounds of WGD at the base of vertebrate evolution \cite{Huminiecki2012}. 

\subsection{Protein structure and function constraints}
% What is protein structure? 
As I described earlier,  all proteins are made of the
same basic constituents, amino acids, which are connected to each other in a certain sequence, also refered to as the primary structure of the protein. The process of forming a stable three-dimensional conformation by a protein is called protein folding. 
 Some of the local arrangements of amino acid residues have been observed in a variety of different proteins.  Combinations of such conformations represent the secondary structure of a given protein.  There are two main kinds of secondary structure: spiral conformations where the amino acids are packed tightly together, called $\alpha$-helices, and long flat sheets, extended in a way that the amino acids are stretched out as far from each other as they can be. Each extended chain is called a $\beta$-strand, and two or more $\beta$-strands held together are called a $\beta$-sheet.

\begin{figure}[t]
\begin{center}
\label{img:populations}
\includegraphics[width=0.7\textwidth]{figures/protein_structure.png}
\end{center}
\caption{Schematic illustration of protein structure. Amino acid sequence forming the primary structure (panel a), secondary structure on the panel b), where the $\alpha$-helix is colored in green and the beta-sheet in blue; on the panel c) secondary structure elements for a three-dimensional structure of a protein, - tertiary structure.} 
\end{figure}

% structural constraints on protein evolution
The selective pressures that might increase or reduce the benefits of acquired traits in a population shape the evolution of animal phenotypes. Selective pressures also exist on the genome level (gene copy numbers, genome size) and on molecular level. The evidence for selective pressures acting on the protein level includes the fact that three-dimensional structure tends to be more conserved than the corresponding amino acid sequence\cite{Illergard2009}.  Many not related proteins demonstrate a high degree of structural conservation even though their primary sequences diverge significantly, suggesting that maintenance of structural folds is a strong constraint on protein evolution \cite{Sousounis2012}. 


% functional constraints 
Additional constraints are added by protein-protein interactions \cite{Park2001} and differences in mutation rate for different proteins. For example, highly expressed proteins are constrained to have fewer mutations to avoid the cost of misfolding \cite{Subramanian2004}. An understanding of the constraints on protein sequence variation is essential for understanding the mechanisms of protein evolution and need to be integrated in all further analysis. 

% protein domains
Most eukaryotic proteins consist of structural domains, each comprising a segment of the polypeptide chain corresponding to one or more elements of secondary structure. Each structural domain folds independently and can even perform an independent function. Protein domains can be also seen from the evolutionary perspective as independent units of protein evolution. When studying duplication and rearrangement events in protein sequences it is important to keep these structural components in mind. Duplication of a whole protein domain might result in making a protein product more stable and put it under a more relaxed evolutionary pressure. The copy of the domain can change over time leading to a modified structure and protein activity. This evolutionary mechanism is particularly common in Metazoa and more often occurs at the N and C-terminals with the exception of protein domain repeats \cite{Bjorklund2006,Bjorklund2005}, like in titin \cite{Higgins1994} and nebulin \cite{Bjorklund2010} proteins. 

%domain shuffling
 Domain shuffling is another result of a rearrangement event in the middle of a structural domain, it occurs when genes coding for different structural domains are joined  to form a new sequence coding for a hybrid protein, such an event can also lead to a novel combination of protein structure and function. Genes that are highly expressed in chordate structures, such as the endostyle, Reissner's fiber of the neural tube, and the notochord show evidence of domain shuffling during their evolution \cite{Kawashima2009}.
 
\section{Protein disorder}
A notion that protein requires a rigid 3D structure to function has been prevalent ever since 1931, when it was shown that protein denaturation leads to a complete loss of function \cite{Wu1995}. This view has been somewhat challenged in the past 30 years. In 1970s the first unstructured, but functional region have been identified in fibrinogen; this region plays a key role in blood clotting \cite{Doolittle1973}.  Nevertheless, up to the 1990s most of the scientists were convinced that disordered regions they could observe by NMR spectroscopy were nothing more than artifacts. The structure of p21, discovered by Kriwacki\cite{Kriwacki1996}  to be fully disordered was something one could not deny any longer; p21 was still able to perform its critical regulatory function although its amino acids assumed a stable conformation only when bound to a kinase. 
 
Several experimental approaches to identify intrinsically disordered regions in proteins have since been developed or adapted from the existing ones. In NMR chemical shifts analyses the pulse sequences have been tailored for resolving chemical shifts in the unfolded state, TROSY and CRINEPT-TROSY NMR spectroscopy now allow identifying the disorder-order transition during protein binding. Temperature and pressure dependent changes in the NMR results can indicate conformational fluctuation of disordered regions. NMR spin relaxation is also advancing to help characterize protein flexibility. H/D exchange in combination with mass spectrometry (HXMS) is used for high-throughput identification of disordered regions, since disordered regions have higher H/D exchange rates to be able to fold before binding with different partners \cite{Balasubramaniam2013}.  

Experimental identification of protein structure and disorder-to-order transitions requires careful sample preparation, expensive equipment and are time consuming.  Here is where computational biology comes to the rescue. Based on the current knowledge of physical properties of protein disorder and amino acid sequences for experimentally identified unstructured regions several methods have been developed to predict disorder from protein sequence \cite{Ferron2006}. On the basis of amino acid composition, hydropathy, capacity of polypeptides to form stabilizing contacts and other differences to known globular proteins these predictors label each amino acid in a protein sequence as ordered or disordered. While using these methods to study protein disorder and its evolution it is important to remember that they are limited to recognize patterns observed in experimentally verified disorder and each predictor is tailored to identify a certain type of characteristics \cite{Dunker2011}.

Roughly 30\% of human proteins are predicted to contain large intrinsically disordered regions \cite{Ward2004}. Many of these proteins perform critical functions in signaling, gene regulation and different protein interactions \cite{Iakoucheva2002}. Disordered regions are in our particular interest for this study also because they lack selective pressures against aggregation and misassembly described earlier, which can makes them subject to a higher mutation rate, possibly giving rise to novel protein functions.   

\mychapter{2}{Materials and Methods}
\section{Databases}
\subsection{FANTOM5 dataset and gene expression tables}
FANTOM5 is the most complete gene expression dataset generated to date, including 952 human and 396 mouse tissues, primary cells and cancer cell-lines, and characterizing transcriptional start sites of all genes in an unbiased fashion, at a single-base nucleotide resolution level. FANTOM consortium is using cap analysis of gene expression (CAGE) for high-throughput identification of sequence tags corresponding to starting sites of the transcripts \cite{Shiraki2003}. CAGE tags are further mapped to RefSeq-annotated gene transcripts and normalized to a certain threshold \cite{Balwierz2009}. Gene expression data of such precision is a powerful source of information for studying protein evolution. 

\subsection{InParanoid}
The InParanoid project gathers proteomes of completely sequenced eukaryotic species plus \emph{Escherichia coli} and provides phylogenetic relationships between them. The new release (7.0) includes 100 species and their collective 1.3 million proteins organized into 42.7 million pairwise ortholog groups \cite{Ostlund2010}. The proteomes are obtained from various sources, including Ensembl, Flybase, NCBI and WormBase. 
\subsection{TreeFam}
TreeFam is a database composed of phylogenetic trees inferred from animal genomes. It provides orthology/paralogy predictions as well the evolutionary history of genes. Curated families are being added progressively, based on seed alignments and trees. Release 1.1, used in the first project, contains curated trees for 690 families and automatically generated trees for another 11,646families \cite{Li2006}. These represent over 128,000 genes from nine fully sequenced animal genomes and over 45,000 other animal proteins from UniProt; approximately 40-85\% of proteins encoded in the fully sequenced animal genomes are included in TreeFam. TreeFam is freely available at \url{http://www.treefam.org}.
\subsection{Disprot}
The DisProt database contains experimentally identified disordered proteins \cite{Sickmeier2007}. Current Disprot release contains 694 with 1539 unstructured regions, 95\% of the proteins are completely disordered. All proteins in the database are annotated with accession numbers from UNIPROT, SWISSPROT, NCBI and other databases. The complete DisProt database is available for download in the FASTA and XML format.
\subsection{MobiDB}
MobiDB's motivation is to build on the content provided by Disprot and other protein structure databases and expand it, with the goal of providing a centralized source for data on disordered regions in protein structures, featuring full coverage of the SwissProt database. Annotations from three disorder predictors (ESpritz, IUPred and DisEMBL) are also included in the database to provide information in cases where experimental annotations are not available. MobiDB includes 26,933 experimentally annotated proteins, 4,662,776 proteins including predicted ones and covers 297 complete proteomes \cite{DiDomenico2012}.
\subsection{STRING database}
The STRING database (\url{http://string-db.org/}) aims to provide a global perspective on protein-protein interactions for as many organisms as feasible \cite{Franceschini2013}. Known and predicted associations are scored and integrated, resulting in comprehensive protein networks covering more than 1100 organisms. The interactions include direct (physical) and indirect (functional) associations; they are derived from four sources: genomic context, high-throughput experiments, conserved coexpression and previous knowledge from text mining of scientific literature.

\section{Computational methods}
\subsection{Sequence search and homology detection.} 
The essential step for studying protein evolution from sequence data is detecting homologous protein sequences. Sequence Similarity Searching is the main method of identifying homology, it involves searching the target sequence in the database of other known proteins and statistically assessing how well the sequences match one another. The most popular sequence similarity search tool is PSI-BLAST. Another approach of inferring homology is implemented by HHblits, general-purpose tool that represents both query and database sequences by profile hidden Markov models (HMMs) (HHblits; \url{http://toolkit.genzentrum.lmu.de/HHblits/}). Compared to PSI-BLAST, HHblits is faster and has 50--100\% higher sensitivity \cite{Remmert2012}.  

\subsection{Multiple sequence alignments (MSA)}
 MSA is generally the alignment of three or more biological sequences (protein or nucleic acid) of similar length. By combining a global MSA with phylogenetic information evolutionary events can be studied in detail. There are many tools available for performing MSA on protein sequences. Clustal Omega is the most recent and rapid MSA tool \cite{Sievers2011} that uses seeded guide trees and HMM profile-profile techniques to generate alignments. It is suitable for large multiple sequence alignments needed for a genome wide evolutionary study. 

\begin{figure}[t]
\begin{center}
\label{img:populations}
\includegraphics[width=0.8\textwidth]{figures/indel.png}
\end{center}
\caption{Identification of insertion and deletion events from a multiple sequence alignment. Given a pair of homologous proteins, A and B, an out-group (C), evolutionary equally distant from them, can be used for distinguishing insertions (green) from deletions (orange)} 
\end{figure}

\subsection{Protein structure prediction}
In-silico protein structure identification includes many approaches of trying to attack the protein folding problem from a computational perspective. One of these methods is  related to studying protein evolution. It involves restricting the search space of possible 3D structures by identifying the key residues that lay in contact with each other in the resulting structure. 

 Recent methodological developments have improved the accuracy of protein contact prediction by modeling the evolutionary couplings as global instead of local statistical models.  Direct coupling analysis (DCA) \cite{Morcos2011} and protein sparse inverse covariance (PSICOV) \cite{Jones2012} establish a global statistical model of the MSA with position-specific variability and interposition coupling. These methods rely on large sets of multiple aligned sequences and assume that correlations between columns in these alignments can be the results of indirect interaction. The new ensemble method for contact prediction PconsC \cite{Skwark2013} combines two predictors based on this approach PSICOV and plmDCA \cite{Ekeberg2013}, and two alignment methods at four different e-value cutoffs and provides an accuracy of more than 70\%. 
 
Information we get from these contact predictions can be extremely useful in identifying structural constraints on protein evolution and key functional sites in flexible disordered regions.

\subsection{Protein disorder predictors}
To date more than fifty methods to identify disorder have been developed \cite{He2009}.  
According to the latest CASP investigation machine learning methods trained on high-resolution X-ray data from PDB (PrDOS and DISOPRED) are the most accurate disorder predictors \cite{Monastyrskyy2011}. Both these methods are quite computationally inefficient though since they require PSI-BLAST homology search information, which is computationally expensive to obtain \cite{Ward2004}. This makes them not always appropriate for large-scale analyses.  We therefore explored other tools for protein disorder predictions. 

  IUPred \cite{Dosztanyi2005} is an algorithm based on the idea that the structural stability of a protein requires a large number of inter-residue interactions. It defines disorder as an inability to form a well-defined 3D structure and assumes that this property is encoded in the protein sequence, because its amino acid composition does not allow sufficient stabilizing interactions to form. 
It implements a position-specific scoring scheme that characterizes the tendency of a given amino acid to fall into an ordered or disordered region. The score is calculated by estimating the total pairwise interaction energy in the neighboring residue of the amino acid. IUPred is a fast and robust tool and offers an advantage compared to machine learning based methods since it is not trained on a limited set of experimentally identified unstructured proteins.  

\mychapter{3}{Projects}
\section{Expression pattern divergence between duplicated genes in mammals}
This section presents the results of a project that was dedicated to exploring the novel FANTOM5 consortium dataset in terms of evolutionary history of mammalian tissues and primary cells. The project status is a completed manuscript by O. Sachenkova and L. Huminiecki, which is currently under review. 
\subsection{Assignment of orthologous tissues between human and mouse}
In analogy to orthologous genes, orthologous tissues can be defined as homologous tissues derived from a common ancestral tissue through the process of speciation.  We developed an ortholog-based interspecies hierarchical clustering, where human and mouse tissue samples are classified based on the protein expression levels of their orthologous genes. We then compared these classes to an intuitive assignment based on samples names. Skin, liver, tongue, heart, pancreas, pituitary gland, thymus and “total RNA control” tissue samples clustered together according to expression pattern. However, the majority of the tissues did not follow the intuitive name based cluster assignment.  This finding suggests that expression pattern evolution rate for protein coding genes is rather rapid, and even on moderate evolutionary distances, such as the human-mouse comparison, interspecies expression pattern differences dominate over phenotype changes on tissue and organ level. 
\subsection{Recently evolved duplicates tend to be tissue-specific with the exception of histones}
We integrated the duplication timing data from the TreeFam database with the FANTOM expression data and explored the evolutionary dynamics of duplicated genes. We confirmed a previously described trend for gradual paralog expression pattern divergence and observed
a novel trend for recently evolved genes to be more tissue-specific in their expression domain.

However, placental mammals (taxon Eutheria) was an outlier to both of the above trends, with an enrichment in housekeeping genes and co-expressed paralogs. Here, three core histone families, H2B, H2A and H3 contribute heavily to the set of highly correlated paralogs, with thymus and testis as the main tissues where paralog histones were highly co-expressed. This finding suggests that that diversification of mammals was accompanied by the expansion of the  histone machinery, with an impact on gene regulation, instead of direct emergence of new protein coding genes. 

\begin{figure}[t]
\begin{center}
\label{img:populations}
\includegraphics[width=0.9\textwidth]{figures/fantom_eutheria.png}
\end{center}
\caption{Expression pattern divergence between paralogs.
Paralogs diverge in expression profiles over evolutionary time, but taxon Eutheria is an outlier to the trend.} 
\end{figure}

\subsection{Phylo-expression signatures}
Phylo-expression signatures were defined as strong associations between individual FANTOM5 samples, and gene duplicates derived from a given taxon. We suggest this way of protein expression analysis as a robust method to study evolutionary history of a given tissue. For example, the emergence of placental mammals was observed to be associated with genes highly expressed in uterus, placenta, and testis.

\subsection{Differences between normal tissues and cancer cell lines}
When paralog analysis was extended with 260 FANTOM5 cancer cell lines, no trend for recently evolved duplicates to be co-expressed could be seen in contrast to the healthy samples. These findings suggest important global differences in regulation of gene expression between normal tissues and primary cells, versus cancer cell lines. As the co-expression signature could be seen in primary cells, something specific to cancer, not the cell culture conditions could be causing this effect.

\begin{figure}[t]
\begin{center}
\label{img:populations}
\includegraphics[width=1\textwidth]{figures/fantom.png}
\end{center}
\caption{Paralog expression divergence in human tissues versus cancer cell-lines.
Distribution of paralog expression distances (Pearson correlation) offered unexpected evidence for global transcriptional deregulation in human cancer cell-lines. Panel (a) represents human tissues, panel (b) human cancer cell-lines} 
\end{figure}

\section{Protein expansion is due to long indels in disordered regions}
This project started with an ambition to explain the length differences between homologous proteins.  At first the dataset to study this problem consisted of proteins from different fungi strains. When I joined the project, it was decided to expand it and select the orthologous proteins more carefully based on phylogenetic relationships between the species. Mammalian, Nematode and Fly dataset have been added to the analysis. 

This project is now complete and the results are published in Molecular Biology and Evolution journal\cite{Light2013}. This section summarizes the findings described in this manuscript. 
\subsection{Most of the length variation in eukaryotic proteins is due to variation in the number of intrinsically disordered residues}
It was observed that the difference in protein length and the difference in number of intrinsically disordered residues are strongly linked in all datasets. The strong coupling between length difference and disorder content is clear regardless of disorder classifier used to predict unstructured regions, and across all ranges of evolutionary distances. Using IUPred and an evolutionary distance of one, i.e. one substitution per residue, a completely ordered protein
shows a length difference of about 3\% while the proteins with more than 30\% disordered residues have a length difference of about 6\%.

\subsection{Disorder begets disorder}
There are two possible explanations for the association between length difference and disorder; either an insertion/deletion event causes a protein to become more disordered or, alternatively, disorder was already a property of the protein region where the insertion occurred prior to an insertion/deletion event. To explore different evolutionary scenarios we formulated three statistical models to explain the expected difference in number of disordered residues, dD, given a length difference of dL. The proximity model was shown to explain a large part of the observation for IUPred (see Figure 6). In this model we assumed that the disorder content of a particular indel is determined by the disorder content of the surrounding residues, i.e. that an unstructured regions tends to grow with evolutionary time. 
\begin{figure}[t]
\begin{center}
\label{img:populations}
\includegraphics[width=0.8\textwidth]{figures/lenvar.png}
\end{center}
\caption{Length differences (dL) compared with differences in the
  number of intrinsically disordered residues (dD) for orthologous
  protein pairs. Every dot is represent a protein pair and the four
  lines represent the observed (black lines) and estimated disorder
  variation using three different models, based on average proteome
  (red), protein (green) or proximity (blue) disorder content.} 
\end{figure}
\subsection{Disorder is prevalent in both insertions and deletions}
Using just pairwise alignments of orthologous proteins it is not possible to explore the differences between insertions and deletions regarding disorder content. To address this problem we used a third species as a phylogenetic out-group and given this out-group assigned an indel to be an insertion or a deletion event. We then studied the distribution of protein disorder in our datasets. In all datasets it could be seen that there are considerably more deletions than insertions, which is consistent with previous studies. Although the deletions are more abundant for all datasets, they are on average considerably shorter than the insertions. As for the disorder content, we show that there are no significant differences between these two types of events. 
Taken together, our findings suggest that indels within disordered protein regions are subject to a relatively low purifying selection compared to ordered proteins. We show that intrinsically disordered regions expand and contract more readily than other regions, but disordered indels mostly occur within already disordered regions. 

\mychapter{4}{Future plans}
\section{Evolution of naturally disordered proteins}
We plan to continue the investigation of mechanisms driving the evolution of disordered regions and the role they play in generating evolutionary novelty in the genome. The first decision we took in this research project was to explore the experimental data available on unstructured proteins instead of working with predicted regions. This could help us identify the evolutionary events with increased certainty and precision. 

Another idea for future analysis is to compare functionally important binding sites in disordered regions and the localization of predicted  structural contacts. This information along with identified evolutionarily conserved sequences could add to our understanding of disorder-to-order transition during protein interactions. It could potentially assist the quantification of mutation rates in unstructured regions, improve multiple sequence alignment and structure prediction techniques for these types of proteins.  

 \section{Evolution of long insertions and deletions in different domains of life}
 Another possible direction for future investigations is focusing on identifying evolutionary patterns of long insertions and deletions in different animal kingdoms, not only in intrinsically disordered regions, but in other structural protein domains as well.  
 
 As described earlier protein domains might undergo substantial changes, for instance significantly increasing or decreasing the length of loops or adding/deleting complete secondary structure elements. Several examples of domains that have been inserted within other domains also exist. These events might explain the existence of regions of protein sequence that cannot be assigned to a homologous domain (orphan protein domains) and represent a novel structural element or function. 
 
 How do these regions evolve and what evolutionary mechanisms drive their development? There are currently two main models of the possible origin of orphan protein domains. The  duplication–-divergence model suggests that newly duplicated genes can go through a phase of fast sequence evolution during which the similarity to founder gene(s) is lost \cite{Orengo2005}. An alternative view proposes the de novo evolution of genes out of random sequences \cite{Tautz2011}. 
 
  To add to this scientific debate a large-scale genome level study should be designed by carefully selecting sequence alignment and homology identification methods and integrating protein structures data. 

\bibliographystyle{ieeetr}
\bibliography{references}

\end{document}

