\documentclass[11pt, a4paper,oneside]{report}
\usepackage[left=3cm,right=3cm,top=3cm,bottom=3cm]{geometry}
\usepackage{setspace}
\onehalfspacing

\newcommand{\mychapter}[2]{
    \setcounter{chapter}{#1}
    \setcounter{section}{0}
    \chapter*{#2}
    \addcontentsline{toc}{chapter}{#2}
}
\usepackage{verbatim}
\usepackage{graphicx}
\usepackage{multirow}
\usepackage{subfigure}
\begin{document}

\title{Evolutionary innovation in eukaryotic proteins}
\author{Oxana Sachenkova}
\date{}
\maketitle

\mychapter{0}{Introduction}
More than 150 years after the publication of Darwin's {\itshape Origin of Species}, we still struggle to understand the processes of evolution and how they shape the diversity of life on Earth. The application of the idea of "descent with modification" to the genome continues to produce fundamental insights into how biological systems evolve. Changes on the DNA level that occur due to recombination events, errors during meiosis and random mutations can change the proteins synthesized in our cells: their structures, expression levels, interactions and their emergent functions.  Analysis of protein evolution is essential for investigating important biological questions such as the evolution of speciation, identification of functionally important sites, drug targets or novel protein functions.
  
Advances in experimental methods such as fast and precise whole-genome sequencing, NMR-spectroscopy and mass spectrometry generate vast amounts of data about protein expression levels, protein structure and function in different organisms.  This large increase of genome-wide information provides new opportunities to study molecular evolution using computational and statistical approaches.

In this research project we are particularly interested in mechanisms of generating evolutionary novelty in protein structure and function. The interplay between gene duplication, insertion and deletion events, mutations and evolutionary constraints is our main focus.  Using computational methods such as protein structure prediction, multiple sequence alignment, machine learning and statistical data analysis we hope to shed light on some aspects of the mechanisms of evolution of novel protein function. Two main research questions I am focusing on are: How do long indels and duplications influence protein function? How do naturally unstructured proteins evolve in comparison with ordered proteins? 

\mychapter{1}{Background}

This chapter introduces the key concepts that are important for understanding the proposed research topic. 
\section{Basic terminology}
\subsection{Building blocks of life}
Cell is the smallest unit of living matter: it consumes energy to maintain its organization and is capable of reproducing itself. These properties are central to the definition of life. All known living organisms consist of one or multiple cells, and in all cases, the whole organism is generated by cell divisions from a single cell. The single cell, therefore, contains all the necessary information and machinery that defines the organism.  Since the discovery of Watson and Crick in 1953\cite{Watson1974} we know that all the living cells store this information in the form of double-stranded molecules of DNA - long unbranched paired polymer chains, formed always of the same four types of monomers, nucleic acids. These monomers, referred to as A, T, C and G are attached  in a linear sequence that encodes the hereditary information of the cell. Information stored in DNA is read out and put to use through a two step process: first, in transcription, segments of DNA sequence are used to guide the synthesis of many molecules of RNA, then, in translation some of the RNA molecules are used to synthesize the molecules of protein and some are helping with this process
\cite{rRNA}. A certain set of rules for translating DNA to protein is shared between all of the living cell.  Different combinations of three base pairs in the DNA, named codons, code for a certain amino acid or serve as a signal of a start or end of a sequence.  This genetic code rule set is redundant, some codons can code for the same amino acid\cite{Turanov2009}. Each segment of DNA that encodes for a functional molecule, either RNA or protein is called a gene\cite{Gerstein2007}.  Complete set of DNA molecules of an organism, including all of its genes is called a genome.  
 
Protein molecules are long unbranched polymer chains, but they are formed by different monomeric building units, amino acids. Each protein molecule, or polypeptide, is created by joining amino acids in a particular sequence and then folding into a three-dimensional form with reactive sites on its surface. Proteins constitute more than 50\%  of the cell's dry mass\cite{Alberts2007} and perform nearly all the cellâ€™s functions: catalyzing metabolic reactions, replicating DNA, responding to stimuli, and transporting molecules from one location to another.

\subsection{Three domains of life}

\subsection{Modern Theory of Evolution}

\section{Evolution of protein coding genes}

\subsection{Protein sequence variations}
\subsubsection{Mutations}
\subsubsection{Gene duplication}

\subsection{Protein structure and function constraints}

\section{Protein disorder}

\mychapter{2}{Materials and Methods}
\section{Databases}
\section{Computational methods}
 
\mychapter{3}{Projects}
\section{Expression pattern divergence between duplicated genes in mammals}

\section{Protein expansion is due to long indels in disordered regions}


\mychapter{4}{Future plans}
\section{Evolution of naturally disordered proteins}
 \section{Evolution of long insertions and deletions in different domains of life}

\bibliographystyle{abbrv}
\bibliography{references}

\end{document}

